### DISCUSSION TITLE
Specify exact image version (e.g. 20230604.1) in configuration file

### DISCUSSION THREAD

**Discussion by lgenzelis** [actions/virtual-environments#7697] [Image configuration]
Hi! In my .yml configuration file I have something like

```
jobs:
  ci:
    runs-on: ubuntu-latest
```

Yesterday, running the pipeline showed this:
![image](https://github.com/actions/runner-images/assets/4582440/b8a6546c-dfe1-4eae-afa9-2da5e7be0d3c)

Today my runs are grabbing the previous version:
![image](https://github.com/actions/runner-images/assets/4582440/67cae6a7-a4dc-4d7e-9019-48418c340e8a)

If I go to the ["Available Images"](https://github.com/actions/runner-images#available-images) of the README, I see the tag from yesterday with that "0.87%", which makes me think that I was in that 0.87% of lucky users yesterday 😅 

<img width="852" alt="Screenshot 2023-06-09 at 2 37 15 PM" src="https://github.com/actions/runner-images/assets/4582440/49a42788-82e3-4f34-8973-194744e49bfc">

So, my question here is, do I have a way to specify the exact version I want to use, in my configuration file? 🙏  I don't know, something like:

```
jobs:
  ci:
    runs-on: ubuntu-latest--20230604.1
```

**Comment by vpolikarpov-akvelon** [comment]
Hey @lgenzelis. You observe this phenomenon because we had to rollback the latest update of Ubuntu runner. So the only version available now is `20230517.1`. It's unlikely that there will ever be possible to choose runner version because GitHub doesn't serve different versions (unless in the middle of update rollout). Sorry for the inconvenience.

You can check actual versions of software installed by selecting appropriate tag instead of main branch: https://github.com/actions/runner-images/blob/ubuntu22/20230517.1/images/linux/Ubuntu2204-Readme.md

**Reply by lgenzelis** [reply]
Got it, thanks for your answer @vpolikarpov-akvelon ! 

> You can check actual versions of software installed by selecting appropriate tag instead of main branch:

Just to be sure, by this you mean I can verify what version is being used for each different piece of software (e.g., to be able to tell whether a runner is using Chrome 113 or 114) by going to the Readme of the corresponding tag. You're **not** telling me that there is a way I can specify the versions I want the runner to use. Correct?

**Reply by vpolikarpov-akvelon** [reply]
Yes, this is correct. The only way to ensure any specific version is to install it manually in your workflow.

Here is, for example, script that is used to install Chrome & Chromedriver to hosted runner image: https://github.com/actions/runner-images/blob/main/images/linux/scripts/installers/google-chrome.sh. You can use the whole script or some parts of it. Moreover you can find this script in runner filesystem but I can't guarantee that it will always be there so I wouldn't rely on this.

Any way, this situation is quite unique. I'd suggest to just wait until the next runner version is rolled out.

**Reply by lgenzelis** [reply]
Thanks for the detailed answer! :) Closing this question.
